test/metabase/models/collection_test.clj

```diff
@@ -1206,6 +1206,13 @@
              Exception
              (db/update! Collection (u/get-id personal-collection) :personal_owner_id (mt/user->id :crowberto)))))))
 
+  (testing "We are not allowed to change the type of a Personal Collection"
+    (mt/with-temp User [my-cool-user]
+      (let [personal-collection (collection/user->personal-collection my-cool-user)]
+        (is (thrown?
+             Exception
+             (db/update! Collection (u/get-id personal-collection) :type "official"))))))
```

Comment:
I think `thrown-with-msg?` would be preferable here because for all we know it's failing because of an NPE or incorrect arity or something unrelated... it's nice to know that it failed because of the condition we wanted to check, and that the correct error message is returned